analyzers/src/SonarAnalyzer.CSharp/Rules/UseValueParameter.cs

```diff
@@ -82,37 +78,22 @@ private static bool IsAccessorValue(IdentifierNameSyntax identifier, SemanticMod
                 return false;
             }
 
-            return semanticModel.GetSymbolInfo(identifier).Symbol is IParameterSymbol parameter &&
-                parameter.IsImplicitlyDeclared;
+            return semanticModel.GetSymbolInfo(identifier).Symbol is IParameterSymbol { IsImplicitlyDeclared: true };
         }
 
         private static string GetAccessorType(AccessorDeclarationSyntax accessorDeclaration)
         {
-            if (accessorDeclaration.IsKind(SyntaxKind.AddAccessorDeclaration) ||
-                accessorDeclaration.IsKind(SyntaxKind.RemoveAccessorDeclaration))
+            if (accessorDeclaration.IsAnyKind(SyntaxKind.AddAccessorDeclaration, SyntaxKind.RemoveAccessorDeclaration))
```

Comment:
You can move this in the `switch` below with `EventDeclarationSyntax` to keep the code uniform